lib/api/model/user/user.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'user.g.dart';
+
+@JsonSerializable()
+class User {
+  User(this.data);
+
+  factory User.fromJson(Map<String, dynamic> json) => _$UserFromJson(json);
+
+  UserData data;
```

Comment:
jeżeli pola są wymagane sugerowałbym `JsonKey(required: true, disallowNullValue: true)` żeby później nie walczyć z nullami w polach nieopcjonalnych w fazie późniejszej niż parsowanie

plus jeżeli by tak zrobić to w konstruktorze te wymagane warto oznaczyć jako `@required` i dodać `assert`